clientServiceClient/api.go

```diff
@@ -28,15 +28,23 @@ type ClientGetResponse struct {
 	Client
 }
 
+// ClientGetPublicResponse is the client information from the endpoint /<client_id>/public
+type ClientGetPublicResponse struct {
+	PublicClient
+}
+
 // ClientRegisterRequest captures the information sent to create a client.
 type ClientRegisterRequest struct {
-	RegistrationToken string `json:"token"`
-	Client            struct {
-		Name       string            `json:"name"`
-		Type       string            `json:"type"`
-		PublicKey  map[string]string `json:"public_key"`
-		SigningKey map[string]string `json:"signing_key,omitemtpy"`
-	} `json:"client"`
+	RegistrationToken string             `json:"token"`
+	Client            ClientRegisterInfo `json:"client"`
```

Comment:
@galxy25 looks like you were right about inline structs, initializing them is a pain 😬